### PR TITLE
feat(config): allow local postgres configuration

### LINKED ISSUE #2611: How to update the PostgreSQL config in the local environment?
Hello,

I tried to setup a local environment and copying my hosted database to my local environment but I encountered some kind of memory issues that you can see here when doing `supabase db reset`:

```
Resetting local database...
Recreating database...
Setting up initial schema...
Applying migration 20240813070343_remote_schema.sql...
ERROR: out of shared memory (SQLSTATE 53200)
At statement 843: SECURITY LABEL FOR "pgsodium" ON COLUMN "public"."abc"."cdf" IS 'ENCRYPT WITH KEY ID xyz'
Try rerunning the command with --debug to troubleshoot the error.
```

When I troubleshoot the issue with the `--debug` flag, it tells me to update the `max_locks_per_transaction` value in the PostgreSQL config. By default it's 64 and I would like to upgrade to 512 like I did on my hosted database.

On my hosted database, I was able to do it with the `supabase postgres-config update` command (documented here https://supabase.com/docs/guides/platform/custom-postgres-config) but it does not seem to work locally as it requires a `project-ref` that I don't have on my local environment.

So, I decided to go inside the docker instance for my database, find the postgresql.conf file located in /etc/postgresql but if I update the `max_locks_per_transaction` there and then restart the instance (by doing `supabase stop` then `supabase start`), my changes in the postgresql.conf file are erased.

So my question is: what's the best way to update the PostgreSQL config when using a local environment?

Thank you.

### PR DESCRIPTION
**Pull request by avallete**
## What kind of change does this PR introduce?

- Apply settings set under db.settings to the local running db instance

## What is the current behavior?

Please link any relevant issues here.

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

Closes #2611

### CODE REVIEW

File: internal/db/start/start.go [hunk 1]
```diff
@@ -56,7 +56,6 @@ func NewContainerConfig() container.Config {
 	env := []string{
 		"POSTGRES_PASSWORD=" + utils.Config.Db.Password,
 		"POSTGRES_HOST=/var/run/postgresql",
-		"POSTGRES_INITDB_ARGS=--lc-ctype=C.UTF-8",
```
Comment (sweatybridge) [inline internal/db/start/start.go:59]: is this intentional? we need this to set the same locale as platform db

File: internal/db/start/start.go [hunk 2]
```diff
@@ -81,13 +80,18 @@ func NewContainerConfig() container.Config {
 			Timeout:  2 * time.Second,
 			Retries:  3,
 		},
-		Entrypoint: []string{"sh", "-c", `cat <<'EOF' > /etc/postgresql.schema.sql && cat <<'EOF' > /etc/postgresql-custom/pgsodium_root.key && docker-entrypoint.sh postgres -D /etc/postgresql
+		Entrypoint: []string{"sh", "-c", `
+cat <<'EOF' > /etc/postgresql.schema.sql && \
+cat <<'EOF' > /etc/postgresql-custom/pgsodium_root.key && \
+cat <<'EOF' >> /etc/postgresql/postgresql.conf && \
```
Comment (sweatybridge) [inline internal/db/start/start.go:86]: Isn't it problematic to append to `postgresql.conf` on every db restart? the file will keep growing in size. I'm not sure what would be a better solution though

Comment (avallete) [reply internal/db/start/start.go:86]: I don't think `/etc/postgresql` is mounted in a volume, so restart should always start with the same file. I'll have a look to be sure.

File: internal/db/start/start.go (hunk 1 above)
Comment (avallete) [reply internal/db/start/start.go:59]: Yeah this is override down the line by an if/else statement: https://github.com/supabase/cli/pull/2796/files/de420f3044ea89994162aea89953559299968ba5#diff-f635ef2693ae7184092f49b3ed3c642a7d3bf0e727de4ff03ea207591c1ef0c2R62-R73

So seemed to me that this doesn't apply.

File: internal/db/start/start.go (hunk 2 above)
Comment (sweatybridge) [reply internal/db/start/start.go:86]: thanks for verifying

File: pkg/config/db.go [hunk 3]
```diff
@@ -146,6 +150,62 @@ func (a *settings) fromRemoteConfig(remoteConfig v1API.PostgresConfigResponse) s
 	return result
 }
 
+// create a valid string to append to /etc/postgresql/postgresql.conf
+func (a *settings) ToPostgresConfig() string {
```
Comment (sweatybridge) [inline pkg/config/db.go:154]: I think we can implement a shortcut here 

```
const pgConfHeader = "# supabase [db.settings] configuration\n"

// create a valid string to append to /etc/postgresql/postgresql.conf
func (a *settings) ToPostgresConfig() string {
	data, _ := ToTomlBytes(*a)
	body := bytes.ReplaceAll(data, []byte{'"'}, []byte{'\''})
	return pgConfHeader + string(body)
}
```

**Review by sweatybridge — APPROVED** [review]

**Pull request merged**